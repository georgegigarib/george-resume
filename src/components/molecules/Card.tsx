import { CardBody } from '@/components/atoms/CardBody'
import { CardContainer } from '@/components/atoms/CardContainer'

interface Props {
  content: React.ReactNode
}

export default function Card({ content = '' }: Props) {
  return (
    <CardContainer className="inter-var">
      <CardBody className="bg-gray-50 group/card  dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl  border">
        {content}
      </CardBody>
    </CardContainer>
  )
}
