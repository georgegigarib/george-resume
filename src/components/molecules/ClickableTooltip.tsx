import React, { useState } from 'react'
import clsx from 'clsx'
import { useIsMobile } from '@/hooks/useIsMobile'
import { ClickAwayListener, IconButton, Tooltip } from '@mui/material'

interface ClickableTooltipProps {
  text: React.ReactNode
  children: React.ReactNode
  bgColor?: string
  textColor?: string
}

export const ClickableTooltip: React.FC<ClickableTooltipProps> = ({
  text = '',
  children,
  bgColor = '',
  textColor = '',
}) => {
  const isMobile = useIsMobile()
  const [open, setOpen] = useState(false)
  const handleTooltipClose = () => {
    setOpen(false)
  }

  const handleTooltipOpen = () => {
    setOpen(true)
  }

  return (
    <ClickAwayListener onClickAway={handleTooltipClose}>
      <div className="ml-1">
        <Tooltip
          data-testid="tooltip"
          PopperProps={{
            disablePortal: true,
          }}
          placement={isMobile ? 'bottom' : 'top'}
          onClose={handleTooltipClose}
          open={open}
          disableHoverListener
          disableTouchListener
          enterTouchDelay={800}
          slotProps={{
            tooltip: {
              sx: {
                color: textColor,
                backgroundColor: bgColor,
              },
            },
          }}
          title={
            <div className={clsx('overflow-auto max-h-[140px] rounded-lg p-2 w-full description-scrollbar')}>
              <h2>{text}</h2>
            </div>
          }
        >
          <IconButton onClick={handleTooltipOpen} size="small">
            {children}
          </IconButton>
        </Tooltip>
      </div>
    </ClickAwayListener>
  )
}
