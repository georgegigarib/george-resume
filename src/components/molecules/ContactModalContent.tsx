import { useRef } from 'react'
import { useTranslation } from 'react-i18next'
import GoogleIcon from '@mui/icons-material/Google'
import MicrosoftIcon from '@mui/icons-material/Microsoft'
import ContentCopyIcon from '@mui/icons-material/ContentCopy'
import { Tooltip } from '@mui/material'

interface ContactModalContentProps {
  message: string
  setMessage: (message: string) => void
  handleGmail: () => void
  handleOutlook: () => void
  handleCopyEmail: () => void
}

export default function ContactModalContent({
  message,
  setMessage,
  handleGmail,
  handleOutlook,
  handleCopyEmail,
}: ContactModalContentProps) {
  const { t } = useTranslation()
  const textareaRef = useRef<HTMLTextAreaElement>(null)

  return (
    <div className="z-40 h-full w-full bg-white dark:bg-black p-6 rounded-xl">
      <h2 className="text-2xl font-bold mb-4 text-black dark:text-white">{t('contact.title')}</h2>
      <div className="flex flex-col h-[calc(100%-80px)]">
        <textarea
          ref={textareaRef}
          value={message}
          onChange={(e) => setMessage(e.target.value)}
          placeholder={t('contact.messagePlaceholder')}
          className="
          flex-grow
          p-4
          rounded-xl
          bg-gray-100 dark:bg-gray-800
          text-black dark:text-white
          resize-none
          focus:outline-none
          focus:ring-2
          focus:ring-blue-500
          mb-4
        "
          required
        />
        <div className="flex gap-3">
          <Tooltip title="Enviar por Gmail" arrow>
            <button
              type="button"
              onClick={handleGmail}
              className="
              flex-1
              bg-red-500
              hover:bg-red-600
              text-white
              py-3
              px-6
              rounded-xl
              font-bold
              flex
              items-center
              justify-center
              gap-2
              transition-all
              duration-300
              transform
              hover:scale-105
            "
            >
              Gmail
              <GoogleIcon />
            </button>
          </Tooltip>

          <Tooltip title="Enviar por Outlook" arrow>
            <button
              type="button"
              onClick={handleOutlook}
              className="
              flex-1
              bg-blue-500
              hover:bg-blue-600
              text-white
              py-3
              px-6
              rounded-xl
              font-bold
              flex
              items-center
              justify-center
              gap-2
              transition-all
              duration-300
              transform
              hover:scale-105
            "
            >
              Outlook
              <MicrosoftIcon />
            </button>
          </Tooltip>

          <Tooltip title="Copiar correo" arrow>
            <button
              type="button"
              onClick={handleCopyEmail}
              className="
              bg-gray-500
              hover:bg-gray-600
              text-white
              py-3
              px-6
              rounded-xl
              font-bold
              flex
              items-center
              justify-center
              gap-2
              transition-all
              duration-300
              transform
              hover:scale-105
            "
            >
              <ContentCopyIcon />
            </button>
          </Tooltip>
        </div>
      </div>
    </div>
  )
}
