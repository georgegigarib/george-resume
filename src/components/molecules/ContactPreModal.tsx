import { useDispatch } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { Tooltip } from '@mui/material'
import EmailIcon from '@mui/icons-material/Email'
import { setContactModalState } from '@/store/modal/modalStatus'

const ContactPreModal = () => {
  const { t } = useTranslation()
  const dispatch = useDispatch()

  const openModal = () => {
    dispatch(setContactModalState(true))
  }

  return (
    <div onClick={openModal} className="cursor-pointer transform hover:scale-105 transition-all duration-300">
      <div
        className="
        bg-gradient-to-r from-blue-500 to-purple-500
        hover:from-purple-500 hover:to-blue-500
        rounded-xl p-4 shadow-lg
        flex flex-col  justify-between
        border-2 border-transparent hover:border-white
        text-white
        transition-all duration-100
      "
      >
        <h2 className="text-lg font-bold">{t('contact.title')}</h2>
        <div className="flex items-center justify-start md:justify-between">
          <p className="text-sm opacity-90 md:w-24">{t('contact.letsWork')}</p>
          <Tooltip title={t('contact.sendMeAMessage')} arrow>
            <span
              className="
            bg-white bg-opacity-20
            hover:bg-opacity-30
            rounded-full
            p-1
            md:p-2
            h-auto
            flex justify-center items-center
            transition-all duration-300
            md:text-xl
            "
            >
              <EmailIcon fontSize="inherit" />
            </span>
          </Tooltip>
        </div>
      </div>
    </div>
  )
}

export default ContactPreModal
