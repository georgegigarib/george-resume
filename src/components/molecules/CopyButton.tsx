import { Tooltip } from '@mui/material'
import { useTranslation } from 'react-i18next'
import ContentCopyIcon from '@mui/icons-material/ContentCopy'
import { useState } from 'react'

interface CopyButtonProps {
  content: string
  tooltipMessage: string
}

export default function CopyButton({ content = '', tooltipMessage = '' }: CopyButtonProps) {
  const [contentCopied, setContentCopied] = useState(false)

  const { t } = useTranslation()

  const handleCopy = () => {
    navigator.clipboard.writeText(content)
    setContentCopied(true)
    setTimeout(() => setContentCopied(false), 1200)
  }

  return (
    <Tooltip title={contentCopied ? t('contact.buttons.copied') : tooltipMessage} arrow>
      <button
        onClick={handleCopy}
        className={`rounded-full p-2 text-sm transition-all duration-300 flex justify-center ${contentCopied ? 'bg-blue-500' : ' hover:bg-gray-300 dark:hover:bg-gray-600'}`}
      >
        <ContentCopyIcon
          fontSize="inherit"
          className={contentCopied ? 'text-white' : 'text-gray-700 dark:text-gray-300'}
        />
      </button>
    </Tooltip>
  )
}
