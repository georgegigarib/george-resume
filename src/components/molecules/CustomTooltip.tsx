import React from 'react'
import { Tooltip, TooltipProps } from '@mui/material'
import clsx from 'clsx'

const CustomTooltip = ({
  title,
  bgColor = 'bg-gray-700',
  textSize = 'text-md',
  className = '',
  children,
  ...props
}: TooltipProps & { bgColor?: string; textSize?: string; className?: string }) => {
  const stopPropagation = (e: React.MouseEvent) => {
    e.stopPropagation()
  }

  return (
    <Tooltip
      title={
        <div onClick={stopPropagation} className={clsx('overflow-auto max-h-[100px] rounded-md', bgColor, textSize)}>
          {title}
        </div>
      }
      {...props}
    >
      <div className={className}>{children}</div>
    </Tooltip>
  )
}

export default CustomTooltip
