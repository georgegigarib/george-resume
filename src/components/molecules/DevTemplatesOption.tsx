import React, { useState } from 'react'
import clsx from 'clsx'
import TechnologyBadge from '@/components/atoms/TechnologyBadge'
import Tooltip from '@mui/material/Tooltip'
import type { Template } from '@/constants/templates'
import { useIsMobile } from '@/hooks/useIsMobile'
import { useTranslation } from 'react-i18next'
import LinkIcon from '@mui/icons-material/Link'
import ExpandMore from '@mui/icons-material/ExpandMore'
import ExpandLess from '@mui/icons-material/ExpandLess'
import { motion } from 'framer-motion'

interface DevTemplatesOptionProps {
  option: Template
}

const DevTemplatesOption: React.FC<DevTemplatesOptionProps> = ({ option }) => {
  const [isExpanded, setIsExpanded] = useState(false)
  const { t } = useTranslation()
  const isMobile = useIsMobile()

  const MAX_VISIBLE_BADGES = isMobile ? 1 : 3
  const remainingTools = option.tools.length - MAX_VISIBLE_BADGES

  const handleContainerClick = () => {
    if (!isExpanded && remainingTools > 0) {
      setIsExpanded(true)
    }
  }

  const handleExpand = () => {
    setIsExpanded(true)
  }

  const handleCollapse = () => {
    setIsExpanded(false)
  }

  return (
    <div
      className={clsx('flex w-full rounded-lg border border-gray-300 flex-nowrap mt-4 relative')}
      data-testid="template-option"
    >
      <div className="overflow-hidden max-w-[86%]">
        <motion.div
          initial={{ height: 'auto' }}
          animate={{ height: isExpanded ? 'auto' : '50px' }}
          transition={{ duration: 0.3 }}
          className="whitespace-nowrap shadow-inner flex mt-1 ml-2 flex-col"
          onClick={handleContainerClick}
        >
          <div className="flex flex-wrap">
            {option.tools.map((tool, index) => (
              <div className="my-1 mb-2" data-testid="tec-badge" key={index}>
                <TechnologyBadge tool={tool} size={isMobile ? 'sm' : 'lg'} />
              </div>
            ))}
          </div>

          <div className="overflow-auto text-wrap my-2 max-h-20 description-scrollbar">{t(option.description)}</div>
        </motion.div>

        <motion.div
          className={clsx('absolute top-3 right-2 cursor-pointer')}
          whileHover={{ scale: 1.05 }}
          onClick={() => window.open(option.url, '_blank')}
        >
          <div className="hover:bg-gray-700 rounded-full p-1  flex items-center justify-center">
            <LinkIcon fontSize="medium" />
          </div>
        </motion.div>
      </div>

      {remainingTools > 0 && !isExpanded && (
        <motion.div
          className="absolute -bottom-3 right-3 cursor-pointer bg-app-grey rounded-full"
          whileHover={{ scale: 1.05 }}
          onClick={handleExpand}
        >
          <Tooltip title={t('dev.templates.modal.showAllTools')} arrow>
            <span className="ml-auto cursor-pointer hover:opacity-70 flex items-center">
              <div className="flex items-center justify-center rounded-full text-sm ">
                <span className="text-white rounded-full flex flex-nowrap">
                  <ExpandMore fontSize="medium" />
                </span>
              </div>
            </span>
          </Tooltip>
        </motion.div>
      )}

      {isExpanded && (
        <>
          <motion.div
            className="absolute -bottom-3 right-3 cursor-pointer bg-blue-950 rounded-full"
            whileHover={{ scale: 1.05 }}
            onClick={handleCollapse}
          >
            <Tooltip title="See less" arrow>
              <div className="flex items-center justify-center rounded-full text-white  hover:bg-gray-700 -bottom-4">
                <ExpandLess fontSize="medium" />
              </div>
            </Tooltip>
          </motion.div>
        </>
      )}
    </div>
  )
}

export default DevTemplatesOption
