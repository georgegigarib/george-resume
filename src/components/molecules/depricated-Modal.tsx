import React, { useState, useEffect } from 'react'
import clsx from 'clsx'
import { useIsMobile } from '@/hooks/useIsMobile'
import { centerModalInViewport } from '@/utils/centerModalInViewPort'
import { StoreState } from '@/store'
import { useSelector } from 'react-redux'
import CloseIcon from '@mui/icons-material/Close'
import { motion, AnimatePresence } from 'framer-motion'
import { IconButton } from '@mui/material'

interface ModalProps {
  trigger: React.ReactNode
  children: React.ReactNode
  layoutId: string
  normalWidth?: string
  mobileHeight?: string
  normalHeight?: string
  className?: string
}

const DepricatedModal: React.FC<ModalProps> = ({
  layoutId,
  trigger,
  children,
  normalWidth = '450px',
  mobileHeight = '350px',
  normalHeight = '500px',
  className,
}) => {
  const [isOpen, setIsOpen] = useState(false)
  const [modalPosition, setModalPosition] = useState({ top: 0, left: 0 })
  const isMobile = useIsMobile()
  const isDakTHemeEnabled = useSelector((state: StoreState) => state.darkTheme.isDarkThemeEnabled)

  useEffect(() => {
    if (isOpen) {
      setModalPosition(centerModalInViewport(isMobile))
    }
    const handleResize = () => setModalPosition(centerModalInViewport(isMobile))
    window.addEventListener('resize', handleResize)
    return () => window.removeEventListener('resize', handleResize)
  }, [isOpen, isMobile])

  const handleClose = () => setIsOpen(false)

  const handleOutsideClick = (e: React.MouseEvent<HTMLDivElement>) => {
    if (e.target === e.currentTarget) {
      handleClose()
    }
  }

  return (
    <>
      <motion.div className="h-full w-full cursor-pointer" layoutId={layoutId} onClick={() => setIsOpen(true)}>
        {trigger}
      </motion.div>

      <AnimatePresence mode="popLayout">
        {isOpen && (
          <>
            <motion.div
              transition={{ ease: 'circIn', duration: 0.5 }}
              className="fixed inset-0 bg-black bg-opacity-50 backdrop-blur-sm flex justify-center items-center z-[20]"
              onClick={handleOutsideClick}
              initial={{ opacity: 0 }}
              animate={{ opacity: 1 }}
              exit={{ opacity: 0 }}
            >
              <motion.div
                layoutId={layoutId}
                className={clsx(
                  'relative rounded-xl p-4',
                  `top-[${modalPosition.top}] left-[${modalPosition.left}]`,
                  className
                )}
                style={{
                  width: isMobile ? 'calc(100% - 80px)' : normalWidth,
                  height: isMobile ? mobileHeight : normalHeight,
                  position: 'fixed',
                }}
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                exit={{ opacity: 0 }}
              >
                <div className="absolute top-4 right-4">
                  <IconButton onClick={handleClose} aria-label="close">
                    <CloseIcon
                      fontSize="medium"
                      style={{ fill: `${isDakTHemeEnabled ? 'white' : 'black'}` }}
                      className="p-1 hover:bg-opacity-35 rounded-full dark:bg-white bg-black bg-opacity-15 dark:bg-opacity-15 hover:scale-105"
                    />
                  </IconButton>
                </div>
                {children}
              </motion.div>
            </motion.div>
          </>
        )}
      </AnimatePresence>
    </>
  )
}

export default DepricatedModal
