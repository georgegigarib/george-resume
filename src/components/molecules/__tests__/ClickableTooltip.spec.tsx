import { render, fireEvent } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach, Mock } from 'vitest'
import { ClickableTooltip } from '@/components/molecules/ClickableTooltip'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { useIsMobile } from '@/hooks/useIsMobile'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

describe('ClickableTooltip component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const renderComponent = (isMobile: boolean, props: any) => {
    ;(useIsMobile as Mock).mockReturnValue(isMobile)
    return render(
      <AppTestProvider>
        <ClickableTooltip {...props} />
      </AppTestProvider>
    )
  }

  const defaultProps = {
    text: 'Tooltip text',
    children: <span>Hover me</span>,
    bgColor: 'bg-blue-500',
    textColor: 'text-white',
  }

  it('matches the snapshot for desktop view', () => {
    const { asFragment } = renderComponent(false, defaultProps)

    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view', () => {
    const { asFragment } = renderComponent(true, defaultProps)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the tooltip text on hover', () => {
    const { getByText } = renderComponent(false, defaultProps)
    const button = getByText('Hover me')
    fireEvent.click(button)

    expect(getByText('Tooltip text')).toBeInTheDocument()
  })

  it('renders the tooltip text on click for mobile view', () => {
    const { getByText } = renderComponent(true, defaultProps)
    const button = getByText('Hover me')
    fireEvent.click(button)

    expect(getByText('Tooltip text')).toBeInTheDocument()
  })
})
