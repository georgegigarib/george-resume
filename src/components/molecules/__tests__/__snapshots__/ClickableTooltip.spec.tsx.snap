// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ClickableTooltip component > matches the snapshot for desktop view 1`] = `
<DocumentFragment>
  <div
    class="ml-1"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <span>
        Hover me
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`ClickableTooltip component > matches the snapshot for mobile view 1`] = `
<DocumentFragment>
  <div
    class="ml-1"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <span>
        Hover me
      </span>
    </button>
  </div>
</DocumentFragment>
`;
